### ISSUE TITLE
SqlBuilder cannot parse sql for large queries

### ISSUE THREAD

**Issue by pwelzien** [NorthernLight1/N.EntityFramework.Extensions#97]
I have finally been able to reproduce a problem with large queries which might explain some weird errors. Entity Framework v6.4.4 changes the query plan and SqlBuilder cannot parse the query.
Add the following test case to UpdateFromQueryAsync:

```
        [TestMethod]
        public async Task With_Many_Value()
        {
            var dbContext = SetupDbContext(true);
            var ids = new long[10000]; // CHANGE HERE, number of ids to find
            int rowUpdated = await dbContext.Orders.Where(a => ids.Contains(a.Id)).UpdateFromQueryAsync(a => new Order { Price = 25.30M });
        }
```

The test case will produce the following exception:

  Message: 
```
Test method N.EntityFramework.Extensions.Test.DbContextExtensions.UpdateFromQueryAsync.With_Many_Value threw exception: 
System.Data.SqlClient.SqlException: Incorrect syntax near the keyword 'SET'.
Incorrect syntax near the keyword 'AS'.
```

  Stack Trace: 
```
DbContextExtensionsAsync.UpdateFromQueryAsync[T](IQueryable'1 querable, Expression'1 updateExpression, Nullable'1 commandTimeout, CancellationToken cancellationToken) line 634
UpdateFromQueryAsync.With_Many_Value() line 34
ThreadOperations.ExecuteWithAbortSafety(Action action)
```

Using 10,000 ids will generate the following sql:

```
SELECT 
    [Project1].[C1] AS [C1], 
    [Project1].[C2] AS [C2], 
    [Project1].[C3] AS [C3], 
    [Project1].[C4] AS [C4], 
    [Project1].[C5] AS [C5], 
    [Project1].[C6] AS [C6], 
    [Project1].[C7] AS [C7]
    FROM ( SELECT 
        [Extent1].[Id] AS [C1], 
        [Extent1].[ExternalId] AS [C2], 
        [Extent1].[Price] AS [C3], 
        [Extent1].[AddedDateTime] AS [C4], 
        [Extent1].[ModifiedDateTime] AS [C5], 
        [Extent1].[Trigger] AS [C6], 
        [Extent1].[Active] AS [C7]
        FROM [dbo].[Orders] AS [Extent1]
    )  AS [Project1]
    WHERE ([Project1].[C1] IN (cast(0 as bigint), ...repeated 9998 times..., cast(0 as bigint))) AND ([Project1].[C1] IS NOT NULL)
```

Using 100 ids will generate the following sql:

```
SELECT 
    [Extent1].[Id] AS [Id], 
    [Extent1].[ExternalId] AS [ExternalId], 
    [Extent1].[Price] AS [Price], 
    [Extent1].[AddedDateTime] AS [AddedDateTime], 
    [Extent1].[ModifiedDateTime] AS [ModifiedDateTime], 
    [Extent1].[Trigger] AS [Trigger], 
    [Extent1].[Active] AS [Active]
    FROM [dbo].[Orders] AS [Extent1]
    WHERE [Extent1].[Id] IN (cast(0 as bigint), ...repeated 98 times... cast(0 as bigint))
```

I don't know the exact threshold when to expect Entity Framework to change the query plan but it would nice if SqlBuilder could parse the updated query.

**Comment by PabloFerroDev** [comment]
I know this is old, but SQL Server limit is 2100 parameters

**Comment by NorthernLight1** [comment]
This item will be addressed in the next release. 

@PabloFerroDev: The code does not use parameters but instead relies on the SQL generated from an IQuerable object. I don't think we need to worry about this limitation then?

**Comment by NorthernLight1** [comment]
This issue was addressed in v1.8.3
